### PR TITLE
Switch from Travis to GitHub Actions

### PR DESCRIPTION
**Pull request by stevepiercy**
- Add Python 3.9
- Merge deps and requirements from tox.ini into setup.py, and remove from tox.ini.
- Set minimum coverage percent to 100.  WebTest's coverage has actually been failing for over 7 years.  Oops!  https://travis-ci.org/github/Pylons/webtest/jobs/23066821
- Bump docs tox env to Python 3.9
- Sort reqs and extras in setup.py
- Use consistent indentation in tox.ini

See also #230.

### CODE REVIEW

**Comment by stevepiercy** [comment]
YES! Coverage is now working properly with GitHub actions. See https://github.com/Pylons/webtest/runs/1389849103?check_suite_focus=true#step:5:283

**Comment by stevepiercy** [comment]
I tried to add in macos and windows, but I give up. I'm not deeply familiar with the requirements on these OSes to get the builds to pass. It would have been a nice addition.

Anyway, this PR is now ready for review.

File: tox.ini
Comment (bertjwregeer) [inline tox.ini:18]: This doesn't work on Windows. Instead I'd recommend running `pytest --cov --cov-report=term-missing` or whatever we do in other Pylons projects. (put flags in `setup.cfg`: https://github.com/Pylons/waitress/blob/master/setup.cfg#L68)

Running coverage stand-alone is not necessary.

Comment (bertjwregeer) [reply tox.ini:18]: This will also fix your Windows issues.

Comment (stevepiercy) [reply tox.ini]: Thanks @bertjwregeer! I pushed another commit and we'll see how the builds go. 🤞 

I also wondered why coverage is run on each platform, other than it might be nice information. This information has been ignored for over 7 years in this project, so I reckon I should remove them here and keep the separate coverage environment.

Comment (stevepiercy) [reply tox.ini]: Oops, looks like pytest-cov plugin needs to be installed.

Comment (stevepiercy) [reply tox.ini]: Yay! We got Windows! I just hope it's not contagious.

There's one failing test on the Windows builds, something to do with file uploads in the forms.py. That's fine by me because the point of this PR is to get CI to work.

Comment (stevepiercy) [reply tox.ini]: Hmm, it looks like Windows barfs on the new line in the uploaded file, where it compares `\n` against a line break in the rendered HTML. I'm not sure how to fix this one.

File: tox.ini
Comment (bertjwregeer) [inline tox.ini:18]: You don't need `--cov` here if you have it in `setup.cfg`

**Comment by bertjwregeer** [comment]
The windows errors are due to git automatically changing the line ending on files to match the system it is being checked out on. So on Windows it will silently turn `\n` into `\r\n`.

This issue was also happening in Pyramid when testing on Windows, you can use a `.gitattributes` file to fix this:

https://github.com/Pylons/pyramid/blob/master/.gitattributes

For example.

File: tox.ini
Comment (bertjwregeer) [reply tox.ini]: See my comment on the main PR.

File: tox.ini
Comment (stevepiercy) [reply tox.ini:18]: Yeah, I removed it from setup.cfg in 7c4fce2 when it failed to build in ec385af.

**Comment by stevepiercy** [comment]
> The windows errors are due to git automatically changing the line ending on files to match the system it is being checked out on. So on Windows it will silently turn `\n` into `\r\n`.
> 
> This issue was also happening in Pyramid when testing on Windows, you can use a `.gitattributes` file to fix this:
> 
> https://github.com/Pylons/pyramid/blob/master/.gitattributes

I never would have figured that out. Thank you! Fixed in 8e3a168.

That leaves coverage as the sole failing build, which is fine for this PR.

**Comment by gawel** [comment]
Hi, nice work. Can you explain why it's ok for the coverage to fail ? (logs are no longer available)

**Comment by stevepiercy** [comment]
Coverage has been broken for at least 7 years. This PR exposes the coverage failures that previously [were failing but were reported as passing](https://travis-ci.org/github/Pylons/webtest/jobs/23066821#L487) because there was no [`--fail-under=100`](https://github.com/Pylons/webtest/pull/231/files#diff-ef2cef9f88b4fe09ca3082140e67f5ad34fb65fb6e228f119d3812261ae51449R29) argument in the automated testing.

**Comment by bertjwregeer** [comment]
The logs for the test run are still available (latest on this PR):

```
2020-11-13T22:32:45.5479953Z Name                  Stmts   Miss  Cover   Missing
2020-11-13T22:32:45.5483892Z ---------------------------------------------------
2020-11-13T22:32:45.5484203Z webtest/__init__.py      15      0   100%
2020-11-13T22:32:45.5484535Z webtest/app.py          337      2    99%   271, 450
2020-11-13T22:32:45.5484864Z webtest/compat.py        17      0   100%
2020-11-13T22:32:45.5485228Z webtest/debugapp.py      60      0   100%
2020-11-13T22:32:45.5485575Z webtest/ext.py            3      0   100%
2020-11-13T22:32:45.5486190Z webtest/forms.py        352     12    97%   90, 117, 252, 273, 278, 287, 465, 472, 572-576
2020-11-13T22:32:45.5486549Z webtest/http.py          85      0   100%
2020-11-13T22:32:45.5486922Z webtest/lint.py         225     27    88%   315, 325, 329, 346, 352, 357, 371, 376, 382, 388, 394, 403, 412, 420, 425, 431, 438, 443, 463, 469, 484, 491, 496, 499, 509, 549, 562
2020-11-13T22:32:45.5487509Z webtest/response.py     232      5    98%   456, 485-491
2020-11-13T22:32:45.5487878Z webtest/sel.py            6      0   100%
2020-11-13T22:32:45.5488417Z webtest/utils.py        101      7    93%   41-42, 98, 118, 130, 133, 166
2020-11-13T22:32:45.5488882Z ---------------------------------------------------
2020-11-13T22:32:45.5489123Z TOTAL                  1433     53    96%
2020-11-13T22:32:45.5489691Z Coverage failure: total of 96 is less than fail-under=100
```

**Comment by stevepiercy** [comment]
> The logs for the test run are still available (latest on this PR):

All logs are available under the travis-ci.org domain:

https://travis-ci.org/github/Pylons/webtest/builds

**Comment by gawel** [comment]
I see, thanks. Not sure we'll be back to 100% one day...

**Review by gawel — APPROVED** [review]

**Comment by bertjwregeer** [comment]
Before this is merged, if getting back to 100% is not a requirement, the fail flag should be removed.

**Comment by stevepiercy** [comment]
Instead of its removal, I could change it to `--fail-under=96` to prevent further erosion of coverage. Then as coverage improves through PRs, we can bump it up.

I could also try to steer contributors to WebTest as a good first PR. The upcoming PloneConf sprints could be one opportunity.

@gawel it's your call.

**Comment by gawel** [comment]
@stevepiercy +1 for the 96% option

**Pull request merged**
